docs/framework/concepts/custom_modules.md

```diff
@@ -0,0 +1,55 @@
+---
+short-description: How to write and use a custom module
+...
+# Custom Modules
+
+The framework provides [modules](modules.md) that app creators can combine to create their apps. Now, they can also write and use their own modules using [Gjs](https://gitlab.gnome.org/GNOME/gjs/wikis/Home).
+
+## Writing a Module
+
+A custom module can look as simple as the example below. Note that a `Custom` class is created from the `Module` class, which is the basic requirement. Other requirements could apply depending on what type of module it is or, more importantly, the type of [slot](modules.md) it should go into.
+
+```javascript
+const Gtk = imports.gi.Gtk;
+const Module = imports.app.interfaces.module;
+
+var Custom = new Module.Class({
+    Name: 'Custom.Custom',
+    Extends: Gtk.Label,
+
+    _init(props={}) {
+        props.label = 'A default text';
+        this.parent(props);
+    },
+});
+```
+
+## Including a Module
+
+To include a module, the file must be added under the `/app/js/` prefix in the app's `gresource` file.
+
+```xml
+<?xml version="1.0" ?>
+<gresources>
+    <gresource prefix="/app/js">
+        <file>custom/custom.js</file>
+    </gresource>
+</gresources>
+```
+
+## Using a Module
+
+Finally, to use a module, it must be referenced from the app's [YAML](concepts/yaml_format.md) file.
+
+```yaml
+root:
+  type: Custom.Custom
+```
+
+## A Custom Importer
+
+To access a custom module class, from another custom module, an importer named `imports_custom` is provided.
+
+```javascript
+const Custom = imports_custom.custom.custom;
```

Comment:
This seems like too many customs :stuck_out_tongue: Maybe we should consider a different developer-facing name for "custom modules"? How about "app modules"? "local modules"? Hmm, maybe "custom modules" is best after all...

Reply:
Sure.